### ISSUE TITLE
when doing continuous runs, export stats periodically

### ISSUE THREAD

**Issue by ldemailly** [fortio/fortio#278]
when running with constant traffic, no stats are emitted until "the end" which isn't that useful

it's also related to #134

**Comment by nkuacac** [comment]
expecting this feature。
we encounted a problem. Testing short connection though loadbalancer for 120s， with some connection 'time out' after 4min，the final summary qps is half of the exact qps

**Comment by ldemailly** [comment]
While periodic export for long run is useful, I don't think it is related to:

> we encounted a problem. Testing short connection though loadbalancer for 120s， with some connection 'time out' after 4min，the final summary qps is half of the exact qps

Set the timeout lower?

**Comment by nkuacac** [comment]
If we have periodic export, we can get the real qps.  And you mean the client timeout option -- the 15s default ? 
> While periodic export for long run is useful, I don't think it is related to:
> 
> > we encounted a problem. Testing short connection though loadbalancer for 120s， with some connection 'time out' after 4min，the final summary qps is half of the exact qps
> 
> Set the timeout lower?

**Comment by ldemailly** [comment]
I would try to understand why you are getting timeouts and why that impacts your expected QPS?

Periodically reporting would not change the math; the qps is how many requests were completed (ie finished, including the reply or the timeout) divided by the time; if it's constant whether you have  1200 requests in 20 minutes or 600 requests in 10 minute it's still 1.0 qps - the goal of this feature is to report maybe every few minutes for say a 1h run; it wouldn't change your 120s test.

**Comment by nkuacac** [comment]
Timeouts always happen as the pressure, but just a little connect, the majority finish in time. If  Periodically reporting, Maybe it with show first half period is huge, then little the next half?

**Comment by ldemailly** [comment]
increase the -c if you have a few bad requests blocking the other ones. or figure out why that's happening? but either it's unrelated to this specific issue. thanks.

**Comment by nkuacac** [comment]
|

**Comment by nkuacac** [comment]
The timeout bugs, we will try to figure out. If there is a periodically report, we can find the bug effectively.

**Comment by ldemailly** [comment]
> The timeout bugs, we will try to figure out. If there is a periodically report, we can find the bug effectively.

I agree! in the meantime remember, you can see errors output in the log as they occur

**Comment by ldemailly** [comment]
also now available through fortiotel's traces/spans
